Engine/Component/ComponentScript.h

```diff
@@ -0,0 +1,33 @@
+#ifndef _COMPONENTSCRIPT_H_
+#define _COMPONENTSCRIPT_H_
+
+#include "Component.h"
+#include "UI/Panel/InspectorSubpanel/PanelComponent.h"
+
+#include <unordered_map>
+#include <string>
+
+class Application;
+class Script;
+class ComponentScript :public Component
+{
+public:
+	ComponentScript();
+	ComponentScript(GameObject * owner, std::string script_name);
```

Comment:
Strings should be passed by reference aswell

Reply:
Done